src/_pytest/pytester.py

```diff
@@ -94,21 +94,16 @@ def pytest_configure(config: Config) -> None:
 
 
 class LsofFdLeakChecker:
-    def get_open_files(self):
-        out = self._exec_lsof()
-        open_files = self._parse_lsof_output(out)
-        return open_files
-
-    def _exec_lsof(self):
-        pid = os.getpid()
-        # py3: use subprocess.DEVNULL directly.
-        with open(os.devnull, "wb") as devnull:
-            return subprocess.check_output(
-                ("lsof", "-Ffn0", "-p", str(pid)), stderr=devnull
-            ).decode()
-
-    def _parse_lsof_output(self, out):
-        def isopen(line):
+    def get_open_files(self) -> List[Tuple[str, str]]:
```

Comment:
why do we not handle any errors? Just curious since the original method doesn't seem to have done it either. But, it looks like it is possible for `subprocess.run` to throw at least `CalledProcessError ` and `TimeoutExpired`.

Reply:
Good point! I think it would be good to handle errors from the `lsof` call here.

If the `lsof` command doesn't exist, the `LsofFdLeakChecker` plugin isn't registered at all, so the issue can't be that `lsof` isn't found at all, but it can still return an error itself.

I don't think `TimeoutExpired` can happen because we don't pass `timeout` to `subprocess.run`, but `CalledProcessError` can happen.

Then there's a question what to do if an error occurs. I guess in this case the most reasonable thing is to just skip the check for the particular test, and issue a warning.